vm/src/pyobject.rs

```diff
@@ -1328,6 +1328,11 @@ pub trait PyValue: fmt::Debug + PyThreadingConstraint + Sized + 'static {
     }
 }
 
+pub trait BorrowValue<'a>: PyValue {
+    type Borrowed: 'a + Deref;
```

Comment:
Clever way of getting around a lack of GATs!

Should this look something like:
```rust
    type Target;
    type Borrowed: 'a + Deref<Target = Self::Target> = &'a Self::Target;
```
maybe?

Reply:
I found `type Target` is redundant. It can be replaced with `<Self::Borrowed as Deref>::Target`

```rust
pub trait BorrowValue<'a>: PyValue {
    type Borrowed: 'a + Deref;
    fn borrow_value(&'a self) -> Self::Borrowed;
    fn clone_value(&'a self) -> <Self::Borrowed as Deref>::Target
    where
        <Self::Borrowed as Deref>::Target: Clone,
    {
        self.borrow_value().clone()
    }
}
```